internal/proxy/proxy.go

```diff
@@ -60,6 +60,10 @@ type InstanceConnConfig struct {
 	// AutoIAMAuthN enables automatic IAM authentication on the instance only.
 	// See Config.AutoIAMAuthN for more details.
 	AutoIAMAuthN bool
+
+	// PublicIP tells the proxy to attempt to connect to the db instance's
+	// public IP address instead of the private IP address
+	PublicIP bool
```

Comment:
should I make this `PublicIP *bool` as it is in the Cloud SQL Proxy?

Reply:
A bool is fine. If we need to change it in the future, this is a private interface so that's fine.

Reply:
Scratch that. We need a bool pointer to support this:

- user sets public IP to false globally
- user enables public ip on a per-instance level.

Reply:
i added a test case for the opposite of this as i think it is more likely, 

- public IP enabled globally
- user disables public IP on a per-instance level